parlai/tasks/dialogue_safety/agents.py

```diff
@@ -0,0 +1,533 @@
+            'Else, will partition train data into 90/10 train/valid and '
+            'use the test set as is.',
+        )
+        parser.add_argument(
+            '--balance-data',
+            type='bool',
+            default=False,
+            help='Balances the data so there are equal numbers of OK and NOT '
+            'OK training data',
+        )
+        return parser
+
+    def __init__(self, opt, shared=None):
+        self.fixed_random = random.Random(42)
+
+        self.use_test_set = opt['use_test_set']
+        self.balance_data = opt['balance_data']
+
+        self.data_path = os.path.join(
+            opt['datapath'], 'dialogue_safety', 'wiki-toxic-comments'
+        )
+
+        if shared and 'data' in shared:
+            self.data = shared['data']
+        else:
+            self.build(opt)
+            self._setup_data(opt['datatype'])
+
+        self.label_candidates = [NOT_OK_CLASS, OK_CLASS]
+
+        opt = copy.deepcopy(opt)
+        super().__init__(opt, shared)
+        self.reset()
+
+    def _get_data(self):
+        # useful constants
+        # all of these colors are bolded
+        RESET = '\033[0m'
+        RED = '\033[1;91m'
+        YELLOW = '\033[1;93m'
+        GREEN = '\033[1;92m'
+        BLUE = '\033[1;96m'
+        CYAN = '\033[1;94m'
+        MAGENTA = '\033[1;95m'
+
+        # only use colors if we're outputting to a terminal
+        USE_COLORS = _sys.stdout.isatty()
+        if not USE_COLORS:
+            RESET = RED = YELLOW = GREEN = BLUE = CYAN = MAGENTA = ''
+
+        # generate the rainbow stars
+        rainbow = [RED, YELLOW, GREEN, CYAN, BLUE, MAGENTA]
+        size = 78 // len(rainbow)
+        stars = ''.join([color + '*' * size for color in rainbow])
+        stars += RESET
+
+        if not os.path.exists(self.data_path):
+            os.makedirs(self.data_path)
+        if not os.path.isfile(os.path.join(self.data_path, 'train.csv')):
+            raise RuntimeError(
+                f'\n\n{stars}\nThis data must be downloaded from '
+                '<https://www.kaggle.com/c/jigsaw-toxic-comment-classification-challenge/data>. '
+                '\nIt cannot be automatically downloaded, as one must agree to '
+                'the competition rules outlined on the website before '
+                'gaining access to the data.\n\n'
+                'Once downloaded, please put the data in the following '
+                f'directory: \n{self.data_path}\n{stars}'
+            )
+
+    def data_to_json(self, pd, file_name):
+        response = pd.to_dict('records')
+        with open(os.path.join(self.data_path, file_name), 'w') as f:
+            f.write(json.dumps(response, indent=4))
+
+    def build(self, opt):
+        self._get_data()
+        version = 'v2.0'
+        read_path = self.data_path
+        if not build_data.built(self.data_path, version):
+            print('[building data from : ' + read_path + ']')
+            if build_data.built(self.data_path):
```

Comment:
not sure we need this check - if the data is built, then this will just remove the data (since you are not redownloading anything in this function)

Reply:
this checks if the data exists but NOT the specific version of the data (i.e. the above checks if it has the right version)

Reply:
yeah, i meant for this specific teacher, since it seems that the downloading of this wiki dataset happens outside of ParlAI right?

Reply:
ohhh gotcha yeah. i'll delete